### PR TITLE
LibGfx/JPEGLoader: Make byte_offset() return offset from start of stream

### PR DESCRIPTION
**Pull request by nico**
JPEGStream::byte_offset() now returns an offset relative to the start of the stream, instead of relative to the buffered part.

No behavior change except if JPEG_DEBUG is set.

### CODE REVIEW

**Comment by nico** [comment]
@LucasChollet, am I misunderstanding how this offset is supposed to be used? I wanted to use the `Huffman stream byte offset $x` line to look at the file at that offset in a hex editor, but without this change that wasn't possible.

**Comment by LucasChollet** [comment]
I'm not a fan of adding logic only to implement a debug statement.
If you think that it's worth, go ahead. But if I remember correctly, it's either the decoder is borked enough to fail on the 10 first symbols (and you don't trigger a cache refill), or it works enough to not fail but return garbage and then the byte offset is useless (I have a local patch that dumps everything into a file).

TLDR:
> am I misunderstanding how this offset is supposed to be used?

I don't know, I don't use it

**Comment by nico** [comment]
I tried to use this friendly-named function while working on #22780 where the problems were more than one chunk size into the data.

I do think this is useful and I have another `dbgln_if()` that uses this locally that I'd like to send out. And this isn't in hot code, so computing it is ~free.

I'm also happy to instead remove `byte_offset()` since it's only used in a debug statement if you prefer that. But having a useful-looking method that returns misleading data is a bit confusing :)

LMK!

**Comment by LucasChollet** [comment]
> I tried to use this friendly-named function

My bad on that :sweat_smile:, for sure we should either fix it or remove it.
And if you find it useful, we can keep it.

**Review by ADKaster — APPROVED** [review]
